ragstack/colbert/llamaindex/retriever.py

```diff
@@ -0,0 +1,63 @@
+
+from llama_index.core import QueryBundle
+from llama_index.core.schema import NodeWithScore, TextNode
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.retrievers import BaseRetriever
+from llama_index.core.constants import DEFAULT_SIMILARITY_TOP_K
+from typing import Any, List, Optional
+
+from ..vector_store import ColbertVectorStoreRetriever
+
+
+class ColbertVectorStoreLlamaIndexRetriever(BaseRetriever):
+    """ColBERT vector index retriever.
+
+    Args:
+        retriever (ColbertVectorStoreRetriever): vector store index.
+        similarity_top_k (int): number of top k results to return.
+    """
+
+    def __init__(
+        self,
+        retriever: ColbertVectorStoreRetriever,
+        similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        callback_manager: Optional[CallbackManager] = None,
+        object_map: Optional[dict] = None,
+        verbose: bool = False,
```

Comment:
Is verbose required for Ilama for Colbert? It's an int in Colbert. But I saw LlamaIndex has verbose too

Reply:
yes it is part of the BaseRetriever:

https://github.com/run-llama/llama_index/blob/main/llama-index-core/llama_index/core/base/base_retriever.py#L50